modules/oe_subscriptions_anonymous/tests/src/Kernel/AccessCheckTest.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_subscriptions_anonymous\Kernel;
+
+use Drupal\Core\Routing\RouteMatch;
+use Drupal\entity_test\Entity\EntityTestBundle;
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\node\Entity\Node;
+use Drupal\Tests\flag\Traits\FlagCreateTrait;
+
+/**
+ * Tests the access checker.
+ */
+class AccessCheckTest extends KernelTestBase {
```

Comment:
This type of test, using directly the access check itself, is more suitable for testing an access check class rather than access in a specific route.
If I replace the $route with `$route = new \Symfony\Component\Routing\Route('/anything/{subscription_id}');`, the test is still green.
It doesn't really test that our route has the access check set, it only tests that the route contains `{subscription_id}`.
Let's change the test to use \Drupal\Core\Access\AccessManager::checkNamedRoute(). This allows to run the checks directly on the real route.

Reply:
Test changed to use access_manager. I had to apply a patch to get the test working: https://www.drupal.org/project/drupal/issues/3158130